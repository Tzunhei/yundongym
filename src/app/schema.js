import path from 'path';
import { mergeTypes, mergeResolvers, fileLoader } from 'merge-graphql-schemas';
import { makeExecutableSchema } from 'apollo-server';
import { IsAuthenticatedDirective, HasRoleDirective } from './directives';

const typesArray = fileLoader(path.join(__dirname, './modules/**/typeDef.js'));
const typeDefs = mergeTypes(typesArray, { all: true });

const resolversArray = fileLoader(
  path.join(__dirname, './modules/**/resolvers.js'),
);
const resolvers = mergeResolvers(resolversArray);

const schema = makeExecutableSchema({
  typeDefs,
  resolvers,
  schemaDirectives: {
    auth: IsAuthenticatedDirective,
    hasRole: HasRoleDirective,
  },
});

export { schema };
