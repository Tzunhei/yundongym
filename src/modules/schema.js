import { makeExecutableSchema } from 'apollo-server';
import { merge } from 'lodash';
import { UserResolvers, UserTypeDef } from './user';

const typeDefs = [UserTypeDef];

const resolvers = merge(UserResolvers);

const schema = makeExecutableSchema({
  typeDefs,
  resolvers,
});

export { schema };
