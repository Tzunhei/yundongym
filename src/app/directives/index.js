export * from './IsAuthenticated';
export * from './HasRole';
