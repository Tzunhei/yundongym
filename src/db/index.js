import Sequelize from 'sequelize';

const {
  DATABASE_USERNAME: username,
  DATABASE_PASSWORD: password,
} = process.env;

const sequelize = new Sequelize('yundongym', username, password, {
  host: 'localhost',
  dialect: 'postgres',
});

export { sequelize };
