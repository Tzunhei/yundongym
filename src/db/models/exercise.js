module.exports = (sequelize, DataTypes) => {
  const Exercise = sequelize.define(
    'Exercise',
    {
      id: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        primaryKey: true,
        allowNull: false,
      },
      name: {
        type: DataTypes.STRING,
        allowNull: false,
        unique: true,
      },
    },
    {},
  );
  Exercise.associate = function ({
    Exercise,
    Set,
    MuscleGroup,
    ExerciseMuscleGroup,
  }) {
    Exercise.hasMany(Set, { foreignKey: 'ExerciseId' });
    Exercise.belongsToMany(MuscleGroup, { through: ExerciseMuscleGroup });
  };

  return Exercise;
};
