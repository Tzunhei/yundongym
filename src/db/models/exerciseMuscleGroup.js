module.exports = (sequelize, DataTypes) => {
  const ExerciseMuscleGroup = sequelize.define(
    'ExerciseMuscleGroup',
    {
      MuscleGroupId: {
        type: DataTypes.UUID,
        allowNull: false,
        references: {
          model: 'MuscleGroup',
          key: 'id',
        },
        onUpdate: 'CASCADE',
        onDelete: 'CASCADE',
      },
      ExerciseId: {
        type: DataTypes.UUID,
        allowNull: false,
        references: {
          model: 'Exercise',
          key: 'id',
        },
        onUpdate: 'CASCADE',
        onDelete: 'CASCADE',
      },
    },
    {
      timestamps: false,
    },
  );

  return ExerciseMuscleGroup;
};
