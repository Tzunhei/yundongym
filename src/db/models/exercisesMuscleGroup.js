module.exports = (sequelize, DataTypes) => {
  const ExerciseMuscleGroup = sequelize.define(
    'ExerciseMuscleGroup',
    {
      muscleGroupId: {
        type: DataTypes.UUID,
        allowNull: false,
        references: {
          model: 'MuscleGroups',
          key: 'id',
        },
        onUpdate: 'CASCADE',
        onDelete: 'CASCADE',
      },
      exerciseId: {
        type: DataTypes.UUID,
        allowNull: false,
        references: {
          model: 'Exercise',
          key: 'id',
        },
        onUpdate: 'CASCADE',
        onDelete: 'CASCADE',
      },
    },
    {},
  );

  return ExerciseMuscleGroup;
};
