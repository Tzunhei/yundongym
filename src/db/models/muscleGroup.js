module.exports = (sequelize, DataTypes) => {
  const MuscleGroup = sequelize.define(
    'MuscleGroup',
    {
      id: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        primaryKey: true,
        allowNull: false,
      },
      name: {
        type: DataTypes.STRING,
        allowNull: false,
      },
    },
    {
      timestamps: false,
    },
  );
  MuscleGroup.associate = function ({
    MuscleGroup,
    Exercise,
    ExerciseMuscleGroup,
  }) {
    MuscleGroup.belongsToMany(Exercise, {
      through: ExerciseMuscleGroup,
    });
  };

  return MuscleGroup;
};
