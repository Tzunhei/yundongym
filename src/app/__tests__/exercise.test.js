import { createTestClient } from 'apollo-server-testing';
import { ApolloServer, gql } from 'apollo-server';
import { schema } from 'app/schema';
import { sequelize } from 'db/models';

const { models } = sequelize;

const GET_ALL_EXERCISES = gql`
  query getExercises {
    getExercises {
      id
      name
      muscleGroups {
        id
        name
      }
    }
  }
`;

const allExercisesMock = [
  {
    id: '1',
    name: 'exercise 1',
    muscleGroups: [
      {
        id: '1',
        name: 'chest',
      },
    ],
  },
];

let server;

beforeAll(() => {
  server = new ApolloServer({
    schema,
    context: () => ({
      headers: 'Bearer 1',
      loggedUser: { id: 1, email: 'test@email.com' },
      models,
    }),
  });
});

describe('query exercises', () => {
  it('queries all exercises', async () => {
    models.Exercise.findAll = jest.fn();
    models.Exercise.findAll.mockReturnValueOnce(allExercisesMock);
    const { query } = createTestClient(server);
    const res = await query({ query: GET_ALL_EXERCISES });

    expect(res.data.getExercises).toEqual(allExercisesMock);
  });
});
