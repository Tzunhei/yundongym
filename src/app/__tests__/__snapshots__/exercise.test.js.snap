// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[QUERY] Exercise get exercise by id 1`] = `
Object {
  "id": "cee739d2-b887-4c5b-8346-ba1c1690bd94",
  "muscleGroups": Array [
    Object {
      "id": "1",
      "name": "chest",
    },
  ],
  "name": "exercise 1",
}
`;

exports[`[QUERY] Exercise queries all exercises 1`] = `
Array [
  Object {
    "id": "cee739d2-b887-4c5b-8346-ba1c1690bd94",
    "muscleGroups": Array [
      Object {
        "id": "1",
        "name": "chest",
      },
    ],
    "name": "exercise 1",
  },
]
`;
